packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -460,6 +468,18 @@ class NavigatorState extends State<Navigator> {
     return _history.length > 1 || _history[0].willHandlePopInternally;
   }
 
+  void didStartUserGesture() { config.observer?.didStartUserGesture(); }
+  void didStopUserGesture() { config.observer?.didStopUserGesture(); }
+
+  AnimationController get currentRouteController {
```

Comment:
This kind of pattern, where two objects can end up both controlling the same animation controller, has historically been a big source of bugs.